File: ast/expression/sizeof.go
package expression

import (
	"github.com/panda-io/micro-panda/ast/ast"
	"github.com/panda-io/micro-panda/ast/ast_types"
)

type Sizeof struct {
	ExpressionBase
	Target ast.Type
}

func (s *Sizeof) Validate(c ast.Context, expected ast.Type) {
	s.Target = c.ResolveType(s.Target)
	s.Typ = ast_types.TypeU32
	s.Const = true
}

File: app/src/main/java/com/github/panda_io/micro_panda/ast/expression/SizeOf.java
package com.github.panda_io.micro_panda.ast.expression;

import com.github.panda_io.micro_panda.ast.type.Type;
import com.github.panda_io.micro_panda.ast.Context;

public class SizeOf extends Expression {
    public Type target;

    public void validate(Context context, Type expected) {
        context.resolveType(this.target);
        this.type = Type.u32;
        this.constant = true;
    }
}
